src/components/search/AggregateSearchResult/index.js

```diff
@@ -7,17 +7,24 @@ module.exports = React.createClass({
 
   propTypes: {
     url: React.PropTypes.string,
+    result: React.PropTypes.object,
     onSelect: React.PropTypes.func
   },
 
+  handleClick: function(e) {
+    e.preventDefault();
```

Comment:
Cool, so if you pass an onSelect then we prevent navigation... 
Otherwise you've passed a url and expect to navigate.

That feels a little tricky. The intent is kind of confusing: this thing is a link, unless you want to do something when it's clicked, then it's sort of a button.
As an aside: I also like the idea of never using anchors unless you mean to navigate.